pkg/api/controller.go

```diff
@@ -3006,6 +3012,65 @@ func (c *Controller) GetCurrentUser(w http.ResponseWriter, r *http.Request) {
 	writeResponse(w, http.StatusOK, response)
 }
 
+func (c *Controller) sendResetPasswordEmail(email []string, token string) error {
+	err := c.Emailer.SendEmail(email, token, token, []string{})
+	if err != nil {
+		c.Logger.Error("Error sending email")
+		return err
+	}
+	c.Logger.Info("Succefully sent email")
+	return err
+}
+
+func (c *Controller) PasswordForgot(w http.ResponseWriter, r *http.Request, body PasswordForgotJSONRequestBody) {
+	user, err := c.Auth.GetUserByEmail(r.Context(), body.Email)
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err)
+		return
+	}
+	if *user.Email != body.Email {
+		writeError(w, http.StatusNotFound, "No such email")
+		return
+	}
+	secret := c.Auth.SecretStore().SharedSecret()
+	currentTime := time.Now()
+	token, err := GenerateJWT(secret, ResetPasswordAudience, auth.InvalidUserID, body.Email, currentTime, currentTime.Add(tokenExpiryDuration))
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err)
+		return
+	}
+	// TODO (@shimi9276) create template for sending the email with link for reset
+	err = c.sendResetPasswordEmail([]string{body.Email}, token)
+	if err != nil {
+		writeError(w, http.StatusBadRequest, err)
+		return
+	}
+	writeResponse(w, http.StatusOK, err)
+}
+
+func (c *Controller) UpdatePassword(w http.ResponseWriter, r *http.Request, body UpdatePasswordJSONRequestBody) {
+	claims, err := VerifyResetPasswordToken(c.Auth, body.Token)
+	if err != nil {
+		writeError(w, http.StatusForbidden, err)
+		return
+	}
+	if strings.Compare(claims.Id, body.Email) != 0 {
+		writeError(w, http.StatusForbidden, "Unauthorized to switch password for given email")
+		return
+	}
+	p, err := model.HashPassword(body.NewPassword)
+	if err != nil {
+		writeError(w, http.StatusServiceUnavailable, err)
+		return
+	}
+
+	err = c.Auth.UpdatePassword(r.Context(), body.Email, p)
+	if err != nil {
+		writeError(w, http.StatusConflict, err)
+	}
```

Comment:
please pay attention - the previous review comments - you need to return if you response to the call and not continue the code on the happy path.

Reply:
Agreed, missed fixing this one. fixed